pkg/sources/adapter/azureservicebussource/adapter.go

```diff
@@ -180,27 +190,35 @@ func entityPath(entityID *v1alpha1.AzureResourceID) string {
 	}
 }
 
-// namespaceFromEnvironment mimics the behaviour of eventhub.NewHubFromEnvironment
-// by returning a servicebus.NamespaceOption that is suitable for the
+// clientFromEnvironment return a azservicebus.Client that is suitable for the
 // authentication method selected via environment variables.
-func namespaceFromEnvironment(entityID *v1alpha1.AzureResourceID) servicebus.NamespaceOption {
-	return func(ns *servicebus.Namespace) error {
-		// SAS authentication (token, connection string)
-		connStr := connectionStringFromEnvironment(entityID.Namespace, entityPath(entityID))
-		sasErr := servicebus.NamespaceWithConnectionString(connStr)(ns)
-		if sasErr == nil {
-			return nil
+func clientFromEnvironment(entityID *v1alpha1.AzureResourceID) (*azservicebus.Client, error) {
+	// SAS authentication (token, connection string)
+	connStr := connectionStringFromEnvironment(entityID.Namespace, entityPath(entityID))
+	if connStr != "" {
+		client, err := azservicebus.NewClientFromConnectionString(connStr, nil)
+		if err != nil {
+			return nil, fmt.Errorf("creating client with connection string: %w", err)
 		}
+		return client, nil
+	}
 
-		// AAD authentication (service principal)
-		aadErr := servicebus.NamespaceWithEnvironmentBinding(entityID.Namespace)(ns)
-		if aadErr == nil {
-			return nil
-		}
+	// AAD authentication (service principal)
+	cred, err := azidentity.NewDefaultAzureCredential(nil)
+	if err != nil {
+		return nil, fmt.Errorf("unable to create azure credentials: %w", err)
+	}
 
-		return fmt.Errorf("neither Azure Active Directory nor SAS token provider could be built - "+
-			"AAD error: %v, SAS error: %v", aadErr, sasErr)
+	if cred != nil {
```

Comment:
This check means that we could get both `err == nil` _and_ `cred == nil` after calling `azidentity.NewDefaultAzureCredential()`, which seems unlikely. The check looks safe to remove to me, since you shortcut on error above.